libbeat/logp/environment.go

```diff
@@ -0,0 +1,65 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package logp
+
+import "strings"
+
+type Environment int
+
+const (
+	DefaultEnvironment Environment = iota
+	SystemdEnvironment
+	ContainerEnvironment
+	MacOSServiceEnvironment
+	WindowsServiceEnvironment
+	InvalidEnvironment
+)
+
+func (v Environment) String() string {
+	switch v {
+	case DefaultEnvironment:
+		return "default"
+	case SystemdEnvironment:
+		return "systemd"
+	case ContainerEnvironment:
+		return "container"
+	case MacOSServiceEnvironment:
+		return "macOS_service"
+	case WindowsServiceEnvironment:
+		return "windows_service"
+	default:
+		return "<invalid>"
+	}
+}
+
+func ParseEnvironment(in string) Environment {
+	switch strings.ToLower(in) {
+	case "default":
+		return DefaultEnvironment
+	case "systemd":
+		return SystemdEnvironment
+	case "docker", "container", "kubernetes", "k8s":
```

Comment:
Maybe `container` is enough here, and then we don't have to think about adding more cases here if some new container manager or runtime is supported in the future. And users don't have to think about the differences of these environments.